apps/edxapp/vars/all/main.yml

```diff
@@ -69,14 +68,56 @@ edxapp_nginx_lms_port: 8071
 edxapp_nginx_htpasswd_secret_name: "edxapp-htpasswd"
 
 # -- celery/redis
-edxapp_celery_lms_high_priority_queue: "edx.lms.core.high"
-edxapp_celery_lms_default_priority_queue: "edx.lms.core.default"
-edxapp_celery_lms_low_priority_queue: "edx.lms.core.low"
-edxapp_celery_lms_high_mem_queue: "edx.lms.core.high_mem"
 
-edxapp_celery_cms_high_priority_queue: "edx.cms.core.high"
-edxapp_celery_cms_default_priority_queue: "edx.cms.core.default"
-edxapp_celery_cms_low_priority_queue: "edx.cms.core.low"
+# LMS queues
+edxapp_celery_lms_high_priority_queue: &lms_high "edx.lms.core.high"
+edxapp_celery_lms_default_priority_queue: &lms_default "edx.lms.core.default"
+edxapp_celery_lms_low_priority_queue: &lms_low "edx.lms.core.low"
+edxapp_celery_lms_high_mem_queue: &lms_high_mem "edx.lms.core.high_mem"
+
+# CMS queues
+edxapp_celery_cms_high_priority_queue: &cms_high "edx.cms.core.high"
+edxapp_celery_cms_default_priority_queue: &cms_default "edx.cms.core.default"
+edxapp_celery_cms_low_priority_queue: &cms_low "edx.cms.core.low"
+
+# LMS workers
+edxapp_celery_lms_high_priority_worker:
+  name: "lms-high"
+  queues: [*lms_high]
+  replicas: 0
+edxapp_celery_lms_default_priority_worker:
+  name: "lms-default"
+  queues:
+    - *lms_high
+    - *lms_default
+    - *lms_low
+    - *lms_high_mem
+  replicas: 1
+edxapp_celery_lms_low_priority_worker:
```

Comment:
Have you thought about wrapping this in a single `workers` variable. Then you could use a loop in the templates (I'm making the assumption that it's possible to include several definitions in definition file?). Something like:
```yaml
celery_workers:
  edxapp_celery_lms_high_mem_worker:
    name: "lms-high-mem"
    queues: [*lms_high_mem]
    replicas: 0
  edxapp_celery_lms_default_priority_worker:
    name: "lms-default"
    queues:
      - *lms_high
      - *lms_default
      - *lms_low
      - *lms_high_mem
    replicas: 1
```

Reply:
This is interesting but too edxapp-specific IMO. If we break the one file/one object paradigm, this will break many things.

Reply:
:thinking: after-thought: I am not so sure of what I wrote. I need to think a bit more about side-effects, but having a single generic DC template for workers might be neat.

Reply:
Unfortunately it seems that we cannot define multiple objects in a single template (for now):

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 138, in run
    res = self._execute()
  File "/usr/local/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 511, in _execute
    self._task.post_validate(templar=templar)
  File "/usr/local/lib/python2.7/dist-packages/ansible/playbook/task.py", line 260, in post_validate
    super(Task, self).post_validate(templar)
  File "/usr/local/lib/python2.7/dist-packages/ansible/playbook/base.py", line 390, in post_validate
    value = templar.template(getattr(self, name))
  File "/usr/local/lib/python2.7/dist-packages/ansible/template/__init__.py", line 525, in template
    disable_lookups=disable_lookups,
  File "/usr/local/lib/python2.7/dist-packages/ansible/template/__init__.py", line 481, in template
    disable_lookups=disable_lookups,
  File "/usr/local/lib/python2.7/dist-packages/ansible/template/__init__.py", line 717, in do_template
    res = j2_concat(rf)
  File "<template>", line 12, in root
  File "/usr/local/lib/python2.7/dist-packages/ansible/plugins/filter/core.py", line 223, in from_yaml
    return yaml.safe_load(data)
  File "/usr/local/lib/python2.7/dist-packages/yaml/__init__.py", line 93, in safe_load
    return load(stream, SafeLoader)
  File "/usr/local/lib/python2.7/dist-packages/yaml/__init__.py", line 71, in load
    return loader.get_single_data()
  File "/usr/local/lib/python2.7/dist-packages/yaml/constructor.py", line 37, in get_single_data
    node = self.get_single_node()
  File "/usr/local/lib/python2.7/dist-packages/yaml/composer.py", line 43, in get_single_node
    event.start_mark)
ComposerError: expected a single document in the stream
  in "<unicode string>", line 6, column 1:
    apiVersion: v1
    ^
but found another document
  in "<unicode string>", line 102, column 1:
    ---
    ^
```

I'll postpone this improvement and roll back to my previous implementation.